src/app/pages/FastBtcPage/components/Withdraw/AddressForm.tsx

```diff
@@ -46,22 +48,37 @@ export const AddressForm: React.FC = () => {
     [set, value],
   );
 
-  const validate = useCallback(async (adr: string) => {
+  const validateAddr = useCallback(async (adr: string) => {
+    let validationReport,
+      validAddressCheck = false,
+      result;
     setAddressValidationState(AddressValidationState.LOADING);
-    const result = await contractReader.call(
-      'fastBtcBridge',
-      'isValidBtcAddress',
-      [adr],
-    );
+    const checkAddressValidity = validate(adr);
+    result = await contractReader.call('fastBtcBridge', 'isValidBtcAddress', [
+      adr,
+    ]);
+    if (checkAddressValidity && result) {
+      validationReport = getAddressInfo(adr);
+      if (
+        validationReport.network.toLowerCase() ===
+          currentNetwork.toLowerCase() &&
+        validationReport.type.toLowerCase() !== 'p2tr'
```

Comment:
```suggestion
        validationReport.type.toLowerCase() !== AddressType.p2tr
```

Reply:
Use `AddressType` enum from `bitcoin-address-validation` package instead, it's much more cleaner solution than a "magic" string.